src/views/LimitedRestorationExtension.vue

```diff
@@ -271,6 +262,19 @@ export default class LimitedRestorationExtension extends Vue {
       // set applicant orgPerson
       entitySnapshot.orgPersons = this.parseApplicantOrgPerson(applicant)
 
+      this.setEntitySnapshot(entitySnapshot)
+
+      if (!restorationFiling.restoration.expiry) {
+        // New limited restoration extension
+        // Set the previously filed limited restoration in the store.
+        await this.setStateFilingRestoration()
+        // parse draft restoration filing into store
+        this.parseRestorationFiling(restorationFiling)
+      } else {
+        this.parseRestorationFiling(restorationFiling)
+        await this.setStateFilingRestoration()
+      }
```

Comment:
I don't understand this code (Karim?)... why set state filing first then parse it, vs parsing it then setting state filing?

Reply:
Yes. Implemented by @JazzarKarim

Reply:
There was a problem I couldn't solve unless I did this.

The problem was when I set the state filing then parse it, the template was rendering before I was able to set approval type court order file number. So when I try to continue a draft, the text field shows empty since I was getting null. When I parse then set, this fixed the problem. 

The other issue was when I create a new limited restoration extension, we have no expiry date. When I parse then set; in the `parseRestorationFiling` method in `filing-template-mixin.ts`, there was a problem. 
On line 806: https://github.com/bcgov/business-edit-ui/blob/09e0f265f5260a0379a610caf59798d62b7eef13/src/mixins/filing-template-mixin.ts#L806
`this.getStateFilingRestoration?.expiry` was returning null. 

I tried literally everything to fix this issue and the if statement was the only way I was able to work around it. The issue mainly was things being called before actually being set in the store etc.

Reply:
OK, may be a reactivity (or mount) issue that this works around.

1. Please create a Tech Debt ticket to fix this in the future.
2. Please add a comment to this code block quickly explaining the issue (and reference the new ticket number).

Reply:
Sounds good. I'll be creating a new ticket now!

Reply:
There is a dependency between `parseRestorationFiling()` and `setStateFilingRestoration` -- the parsing uses "approvalType" and "expiry" from the state filing.

Therefore these won't be stored if you don't set the state filing first. @JazzarKarim What tests can we run to verify what's broken by this?

Reply:
Is there a scenario where `this.mutateRestorationApprovalType(this.getStateFilingRestoration?.approvalType)` (from Filing Template Mixin) could in fact store an empty (undefined) value?

Reply:
No not any that I'm aware of. I've tried every single possible scenario before merging my PR back then.

Reply:
OK, sounds mitigated for now.

Please post number of new ticket (to fix the root cause) here.

Reply:
New ticket number: https://github.com/bcgov/entity/issues/15862

Reply:
Ketaki, I'll wait for your code comment before resolving this conversation.

Reply:
All good from my side. @JazzarKarim, I will add ticket number as reference as I am modifying the code in this ticket. Please let me know if you would like to add more explanation.

Reply:
Thanks. LGTM.